src/ts/ConvivaAnalytics.ts

```diff
@@ -3,1140 +3,281 @@ import type {
+    this.registerPlayerEvents();
 
-    const deviceMetadataFromConfig = this.config.deviceMetadata || {};
-    const deviceMetadata: Conviva.ConvivaDeviceMetadata = {
-      [Conviva.Constants.DeviceMetadata.CATEGORY]:
-        deviceMetadataFromConfig.category || this.config.deviceCategory || Conviva.Constants.DeviceCategory.WEB,
-      [Conviva.Constants.DeviceMetadata.BRAND]: deviceMetadataFromConfig.brand,
-      [Conviva.Constants.DeviceMetadata.MANUFACTURER]: deviceMetadataFromConfig.manufacturer,
-      [Conviva.Constants.DeviceMetadata.MODEL]: deviceMetadataFromConfig.model,
-      [Conviva.Constants.DeviceMetadata.TYPE]: deviceMetadataFromConfig.type,
-      [Conviva.Constants.DeviceMetadata.VERSION]: deviceMetadataFromConfig.version,
-      [Conviva.Constants.DeviceMetadata.OS_NAME]: deviceMetadataFromConfig.osName,
-      [Conviva.Constants.DeviceMetadata.OS_VERSION]: deviceMetadataFromConfig.osVersion,
+    this.convivaSsaiAnalytics = new ConvivaAnalyticsSsai(this.convivaAnalyticsTracker);
+
+    // Do not expose `reset` method to the public API.
+    this.ssai = {
+      get isAdBreakActive() {
+        return this.convivaSsaiAnalytics.isAdBreakActive;
+      },
+      reportAdBreakStarted: this.convivaSsaiAnalytics.reportAdBreakStarted.bind(this.convivaSsaiAnalytics),
+      reportAdStarted: this.convivaSsaiAnalytics.reportAdStarted.bind(this.convivaSsaiAnalytics),
+      reportAdFinished: this.convivaSsaiAnalytics.reportAdFinished.bind(this.convivaSsaiAnalytics),
+      reportAdSkipped: this.convivaSsaiAnalytics.reportAdSkipped.bind(this.convivaSsaiAnalytics),
+      reportAdBreakFinished: this.convivaSsaiAnalytics.reportAdBreakFinished.bind(this.convivaSsaiAnalytics),
     };
-    Conviva.Analytics.setDeviceMetadata(deviceMetadata);
-
-    let callbackFunctions: Record<string, Function> = {};
-    callbackFunctions[Conviva.Constants.CallbackFunctions.CONSOLE_LOG] = this.logger.consoleLog;
-    callbackFunctions[Conviva.Constants.CallbackFunctions.MAKE_REQUEST] = new Html5Http().makeRequest;
-    const html5Storage = new Html5Storage();
-    callbackFunctions[Conviva.Constants.CallbackFunctions.SAVE_DATA] = html5Storage.saveData;
-    callbackFunctions[Conviva.Constants.CallbackFunctions.LOAD_DATA] = html5Storage.loadData;
-    callbackFunctions[Conviva.Constants.CallbackFunctions.CREATE_TIMER] = new Html5Timer().createTimer;
-    callbackFunctions[Conviva.Constants.CallbackFunctions.GET_EPOCH_TIME_IN_MS] = new Html5Time().getEpochTimeMs;
-
-    const settings: Record<string, string | number> = {};
-    settings[Conviva.Constants.GATEWAY_URL] = config.gatewayUrl;
-    settings[Conviva.Constants.LOG_LEVEL] = this.config.debugLoggingEnabled
-      ? Conviva.Constants.LogLevel.DEBUG
-      : Conviva.Constants.LogLevel.NONE;
-
-    Conviva.Analytics.init(customerKey, callbackFunctions, settings);
-
-    this.contentMetadataBuilder = new ContentMetadataBuilder(this.logger);
-
-    this.registerPlayerEvents();
   }
 
-  /**
```

Comment:
Why did you remove the documentation for all the public methods? 🤔

Reply:
Moved in https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/112/commits/5803fb7253bcc383ceaec4f74a6d89255e1f09e1.